tests/engine/test_debit_note_intervals.py

```diff
@@ -0,0 +1,253 @@
+            _is_none,
+        ),
+        (
+            {"details___ref__demand__properties": {PROP_DEBIT_NOTE_INTERVAL_SEC: 100}},
+            {},
+            10,
+            0,
+            0,
+            _message_matches("^Too many debit notes"),
+            _is_none,
+        ),
+        (
+            {"details___ref__demand__properties": {PROP_PAYMENT_TIMEOUT_SEC: 100}},
+            {"_base__payment_due_date": True},
+            100,
+            0,
+            0,
+            _is_none,
+            _is_none,
+        ),
+        (
+            {"details___ref__demand__properties": {PROP_PAYMENT_TIMEOUT_SEC: 100}},
+            {"_base__payment_due_date": True},
+            10,
+            0,
+            0,
+            _is_none,
+            _message_matches("^Too many payable debit notes"),
+        ),
+        (
+            {"details___ref__demand__properties": {PROP_DEBIT_NOTE_INTERVAL_SEC: 100}},
+            {},
+            300,
+            2,
+            0,
+            _is_none,
+            _is_none,
+        ),
+        (
+            {"details___ref__demand__properties": {PROP_DEBIT_NOTE_INTERVAL_SEC: 100}},
+            {},
+            300,
+            4,
+            0,
+            _message_matches("^Too many debit notes"),
+            _is_none,
+        ),
+        (
+            {"details___ref__demand__properties": {PROP_PAYMENT_TIMEOUT_SEC: 100}},
+            {"_base__payment_due_date": True},
+            300,
+            0,
+            2,
+            _is_none,
+            _is_none,
+        ),
+        (
+            {"details___ref__demand__properties": {PROP_PAYMENT_TIMEOUT_SEC: 100}},
+            {"_base__payment_due_date": True},
+            300,
+            0,
+            4,
+            _is_none,
+            _message_matches("^Too many payable debit notes"),
+        ),
+    ],
+)
+def test_verify_debit_note_intervals(
+    agreement_kwargs,
+    debit_note_kwargs,
+    duration,
+    num_debit_notes,
+    num_payable_debit_notes,
+    vdni_condition,
+    vpt_condition,
+):
+    agreement = AgreementFactory(**agreement_kwargs)
+    debit_note = DebitNoteFactory(_base__agreement_id=agreement.id, **debit_note_kwargs)
+    engine = mock_engine(agreement, debit_note, num_debit_notes, num_payable_debit_notes)
+
+    engine._num_debit_notes[debit_note.activity_id] += 1  # noqa
+    if debit_note.payment_due_date:
+        engine._num_payable_debit_notes[debit_note.activity_id] += 1  # noqa
+
+    assert vdni_condition(engine._verify_debit_note_interval(agreement, debit_note, duration))
+    assert vpt_condition(engine._verify_payment_timeout(agreement, debit_note, duration))
+
+
+@pytest.mark.skipif(sys.version_info < (3, 8), reason="AsyncMock requires python 3.8+")
+@pytest.mark.asyncio
+@pytest.mark.parametrize(
+    "agreement_kwargs, debit_note_kwargs, created_at_cb, "
+    "num_debit_notes, num_payable_debit_notes, "
+    "debit_note_accepted, agreement_terminated",
```

Comment:
Is it not always true that `debit_note_accepted is not agreement_terminated`?
If yes, it would be nice to remove one of them from the params.

Reply:
it's not always true, as the parameters show

Reply:
Argh, OK, I missed the single-line parameters.